prow/jobs/kyma/kyma-vulnerability-scanner.yaml

```diff
@@ -34,4 +34,27 @@ periodics:
         resources:
           requests:
             memory: 100Mi
-            cpu: 50m
\ No newline at end of file
+            cpu: 50m
+- name: kyma-gcp-container-vulnerability-collector
+  decorate: true
+  cron: "0 0 1 * *" # “At 00:00 on day-of-month 1.”
+  labels:
+    preset-sap-slack-bot-token: "true"
+    preset-kyma-slack-channel: "true"
+  spec:
+    containers:
+      - image: eu.gcr.io/kyma-project/prow/test-infra/bootstrap:v20181121-f3ea5ce
+        command:
+          - "bash"
+        env:
+          - name: KYMA_PROJECT_DIR
+            value: "/home/prow/go/src/github.com/kyma-project"
+          - name: SCAN_WHITELIST
+            value: "acs-installer,add-ons-ui,alpine-net,api-controller,apiserver-proxy,application-broker,application-connectivity-certs-setup-job,application-connectivity-validator,application-connector,application-gateway,application-operator,application-proxy,application-registry,ark-plugins,asset-metadata-service,asset-store-controller-manager,asset-upload-service,assetstore-controller-manager,azure-broker,binding-usage-controller,brokers-ui,c4c-mock,catalog,catalog-ui,changelog-generator,cms-controller-manager,commerce-mock,configurations-generator,connection-token-handler,connectivity-certs-controller,connector-service,console,console-backend-service,contacts,contacts-app,contacts-app-ui,contacts-ui-web,contacts-web-ui,content,content-to-json-generator,content-ui,dex,dex-static-user-configurer,documentation-generator,e2etest,environments,etcd-backup,etcd-tls-setup,event-bus,event-bus-e2e-subscriber,event-bus-nats-init,event-bus-publish,event-bus-publish-knative,event-bus-push,event-bus-sub-validator,event-bus-subscription-controller-knative,event-gateway,event-service,example,example-test-buildpack,examples,faros-bundles,faros-mock,gateway,gcp-broker-provider,gcp-installer,golang-buildpack,helm-broker,helm-broker-reposerver,helm-broker-tools,iam-kubeconfig-service,incubator,installer,instances,instances-ui,integration-dex,istio-kyma-patch,istio-webhook,k8s-dashboard-proxy,kube-rbac-proxy,kubebuilder-poc,kubeless-images-node6,kubeless-images-node8,kyma,kyma-installer,kyma-operator,lambda,log-ui,marketing-mock,marketplaces,metadata-service,namespace-controller,navigation-builder,node-buildpack,pod-tracker,prow,proxy-service,rc,release,release-adder,remote-environment-broker,remote-environment-controller,resources-uploader,sbu-upgrade-protector,service-catalog,stability-checker,static-users-generator,test-environments,test-infra,test-logging,test-logging-monitoring,test-namespace-controller,ui-api-layer,varkes-app-connector,varkes-app-connector-client,varkes-commerce-mock,varkes-marketing-mock,varkes-odata-mock,varkes-openapi-mock,velero-plugins,watch-pods,xf-application-mocks,xip-patch"
```

Comment:
Setting the list of components to scan has one drawback, you have to maintain this list. Developers will not know nor remember to update this list when they to component renaming or add/delete component. Instead I suggest fetching components from gcr via API and perform scanning based on that list. Here could be list of pottentially excluded components which should not be skanned (although everything we build and push to our gcr repo should be scanned)

Reply:
In the implementation I'm actually already checking for ALL containers on gcr. It's just too many and I think for this case a whitelist is better than a blacklist. I think when I checked, there's more than 280 containers, while there's actually about 59 component containers. If we can come up with a better way to figure out, which containers we want or don't want to retrieve the vulnerability results from.

Reply:
Well, from security POV all containers used in kyma should be scanned. If there is 280 containers question shoud be asked : do we really need them all? Answer should be provided by kyma team. For some operations it should be unified if possible. Why we should scan all? Simple example might be a container which is a simple job  doing kubectl stuff but using old dependency versions with some vulnerabilities.

Validating what is required and used from the list of 280 containers is a topic for a follow up in kyma repo ofc.

Reply:
Removed whitelisting.